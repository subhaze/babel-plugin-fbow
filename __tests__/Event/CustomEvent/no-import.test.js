const babel = require("babel-core");
const plugin = require("../../../index.js");

const testCode = `
obj.addEventListener("cat", function(e) { process(e.detail) });
var event = new CustomEvent("cat", {
  detail: {
    hazcheeseburger: true
  }
});
obj.dispatchEvent(event);
`;

it("doesn't add CustomEvent constructor import", () => {
	const { code } = babel.transform(testCode, {
		plugins: [plugin],
		sourceType: "script",
	});
	expect(code).toMatchSnapshot();
});
