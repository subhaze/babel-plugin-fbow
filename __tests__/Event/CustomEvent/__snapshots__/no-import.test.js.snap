// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doesn't add CustomEvent constructor import 1`] = `
"
obj.addEventListener(\\"cat\\", function (e) {
  process(e.detail);
});
var event = new CustomEvent(\\"cat\\", {
  detail: {
    hazcheeseburger: true
  }
});
obj.dispatchEvent(event);"
`;
